jxl/src/features/patches.rs

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// TODO(firsching): remove once we use this!
+#![allow(dead_code)]
+
+use crate::{
+    bit_reader::BitReader,
+    entropy_coding::decode::Histograms,
+    error::{Error, Result},
+    frame::DecoderState,
+    util::tracing_wrappers::*,
+};
+
+// Context numbers as specified in Section C.4.5, Listing C.2:
+const NUM_REF_PATCH_CONTEXT: usize = 0;
+const REFERENCE_FRAME_CONTEXT: usize = 1;
+const PATCH_SIZE_CONTEXT: usize = 2;
+const PATCH_REFERENCE_POSITION_CONTEXT: usize = 3;
+const PATCH_POSITION_CONTEXT: usize = 4;
+const PATCH_BLEND_MODE_CONTEXT: usize = 5;
+const PATCH_OFFSET_CONTEXT: usize = 6;
+const PATCH_COUNT_CONTEXT: usize = 7;
+const PATCH_ALPHA_CHANNEL_CONTEXT: usize = 8;
+const PATCH_CLAMP_CONTEXT: usize = 9;
+const NUM_PATCH_DICTIONARY_CONTEXTS: usize = 10;
+
+// Blend modes
+// The new values are the old ones. Useful to skip some channels.
```

Comment:
What about a enum? (and then using `#[derive(FromPrimitive)]`)

(probably also makes sense for the contexts)

Reply:
made both an enum